src/tentaclio/fs/api.py

```diff
@@ -39,6 +40,19 @@ def listdir(url: str) -> Iterable[str]:
     return (str(entry.url) for entry in entries)
 
 
+def copy(source: str, dest: str):
+    """Copy the contents of the origin URL into the dest url.
+
+    Depending on the type of resource this can be much more performant than
+    >>> with tio.open(source) as reader, tio.open(dest) as writer:
+            writer.write(reader.read())
+    """
+    source_auth = credentials.authenticate(source)
+    dest_auth = credentials.authenticate(dest)
+    copier = COPIER_REGISTRY.get_handler(source_auth.scheme + "+" + dest_auth.scheme)
```

Comment:
I think it's safer to use a tuple instead of "magical" string constructs

Reply:
it's not a _magical construct_ per se, all the registries in tentaclio are indexed by the url scheme, and having two schemes appended by a `+` is more than common. i.e. "git+ssh", "athena+rest" 
https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Generic_syntax
so it's a valid url and keeps the consistency across the library.

Reply:
fair enough